common/core/nesteddisplay.cpp

```diff
@@ -22,16 +22,145 @@
 #include <sstream>
 
 #include <hwctrace.h>
+#include <hwclayer.h>
+#include "utils_android.h"
 
 namespace hwcomposer {
+#ifdef NESTED_DISPLAY_SUPPORT
+#define METADATA_BUFFER_SIZE 12000
+#define METADATA_STREAM_START 0xF00D
+#define METADATA_STREAM_END 0xCAFE
+
+static char buf[METADATA_BUFFER_SIZE];
+
+#define HYPER_DMABUF_PATH "/dev/hyper_dmabuf"
+std::unique_ptr<SocketThread> NestedDisplay::st_ = NULL;
+int NestedDisplay::mclient_sock_fd = -1;
+
+SocketThread::SocketThread(int *client, bool *connection, int server)
+    : HWCThread(-8, "SocketThread") {
+  client_sock_fd = client;
+  sock_fd = server;
+  connected = connection;
+  mEnabled = true;
+}
+
+SocketThread::~SocketThread() {
+}
 
-NestedDisplay::NestedDisplay() {
+void SocketThread::Initialize() {
+  if (InitWorker())
+    Resume();
+  else
+    ETRACE("Failed to initalize CompositorThread. %s", PRINTERROR());
+}
+
+void SocketThread::setEnabled(bool enabled) {
+  if (mEnabled != enabled) {
+    mEnabled = enabled;
+    Resume();
+  }
+}
+
+void SocketThread::HandleRoutine() {
+  socklen_t clilen;
```

Comment:
client_length ? can u make this more descriptive ?

Reply:
yes, will modify it.